buildSrc/src/main/java/com/google/firebase/gradle/plugins/FireEscapeArtifactPlugin.kt

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.gradle.plugins
+
+import java.io.File
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.publish.PublishingExtension
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.api.publish.maven.plugins.MavenPublishPlugin
+import org.gradle.api.tasks.bundling.Jar
+import org.gradle.api.tasks.bundling.Zip
+
+class FireEscapeArtifactPlugin : Plugin<Project> {
+  var project: Project? = null
+
+  override fun apply(target: Project) {
+    project = target
+
+    project!!.afterEvaluate {
+      val firebaseLibrary = extensions.findByType(FirebaseLibraryExtension::class.java)
+      val fireEscapeTask =
+        tasks.create("mavenAarFireEscapeArtifact", Zip::class.java) { classifier = "fireescape" }
+      plugins.withType(MavenPublishPlugin::class.java) {
+        extensions.configure(PublishingExtension::class.java) {
+          publications.withType(MavenPublication::class.java) {
+            if ("mavenAar" == name) {
+              configurePublication(this, fireEscapeTask, firebaseLibrary!!.type)
+            }
+          }
+        }
+      }
+    }
+  }
+
+  private fun configurePublication(
```

Comment:
While I'm the biggest supporter of functional practices, I feel this method is not really needed. Especially under the context of registering tasks as individuals in a modular way. You could even abstract it all the way to:
```kotlin
override fun apply(target: Project) {
    target.afterEvaluate {
      val apiTxtFile = registerApiTxtFileTask(project)
      val proguardMappingFile = registerProguardMappingFileTask(project)
      val javadoc = registerJavadocTask(project)

      val zippedArtifact = project.tasks.register<Zip>("maven") {
        from(apiTxtFile)
        if (project.isAndroid()) {
          from(proguardMappingFile)
        }
      }

      plugins.withType<MavenPublishPlugin> {
        extensions.configure<PublishingExtension> {
          publications.getByName<MavenPublication>("mavenAar") {
            artifact(zippedArtifact)
            artifact(javadoc)
          }
        }
      }
    }
  }
```

But I also wouldn't be against this either:

```kotlin
  override fun apply(target: Project) {
    target.afterEvaluate {
      if (supportsMavenPublishing(target)) {
        val apiTxtFile = registerApiTxtFileTask(project)
        val proguardMappingFile = registerProguardMappingFileTask(project)
        val javadoc = registerJavadocTask(project)
        val zippedArtifact = registerArtifactTask(project, apiTxtFile, proguardMappingFile)

        configurePublication(project, zippedArtifact, javadoc)
      }
    }
  }

  private fun supportsMavenPublishing(project: Project) =
    project.plugins.hasPlugin(MavenPublishPlugin::class.java)

  private fun configurePublication(
    project: Project,
    zippedArtifact: TaskProvider<Zip>,
    javadoc: TaskProvider<Jar>
  ) {
    project.extensions.configure<PublishingExtension> {
      publications.getByName<MavenPublication>("mavenAar") {
        artifact(zippedArtifact)
        artifact(javadoc)
      }
    }
  }
```